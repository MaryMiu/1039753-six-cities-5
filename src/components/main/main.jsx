import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import Header from "../header/header";
import PlacesList from "../places-list/places-list";
import Map from "../map/map";
import Menu from "../menu/menu";
import {connect} from "react-redux";

class Main extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {city, offers} = this.props;
    const coord = offers.map((offer) => offer.coord);
    const mapStyle = {
      display: `flex`,
      height: `100%`,
      width: `100%`
    };
    const currentClasses = {
      listClass: `cities__places-list tabs__content`,
      cardClass: `cities__place-card`,
      imgClass: `cities__image-wrapper`,
    };

    return (
      <div className="page page--gray page--main">
        <Header />
        <main className="page__main page__main--index">
          <h1 className="visually-hidden">Cities</h1>
          <div className="tabs">
            <Menu />
          </div>
          <div className="cities">
            <div className="cities__places-container container">
              <section className="cities__places places">
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">{offers.length} places to stay in {city}</b>
                <form className="places__sorting" action="#" method="get">
                  <span className="places__sorting-caption">Sort by</span>
                  <span className="places__sorting-type" tabIndex="0">
                  Popular
                    <svg className="places__sorting-arrow" width="7" height="4">
                      <use xlinkHref="#icon-arrow-select"></use>
                    </svg>
                  </span>
                  <ul className="places__options places__options--custom places__options--opened">
                    <li className="places__option places__option--active" tabIndex="0">Popular</li>
                    <li className="places__option" tabIndex="0">Price: low to high</li>
                    <li className="places__option" tabIndex="0">Price: high to low</li>
                    <li className="places__option" tabIndex="0">Top rated first</li>
                  </ul>
                </form>
                <PlacesList offers={offers} currentClasses={currentClasses} />
              </section>
              <div className="cities__right-section">
                <section className="cities__map map">
                  <Map coord={coord} mapStyle={mapStyle} />
                </section>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  city: state.city,
  offers: state.offers,
});

Main.propTypes = {
  city: PropTypes.string.isRequired,
  offers: PropTypes.array.isRequired,
};

export {Main};
export default connect(mapStateToProps)(Main);

