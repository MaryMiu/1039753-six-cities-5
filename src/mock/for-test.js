export const offers = [{
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
  ],
  title: `The house among olive `,
  rating: 3.8,
  type: `HOUSE`,
  bedrooms: 3,
  price: 493,
  goods: [
    `Laptop friendly workspace`,
    `Washer`,
    `Towels`,
    `Fridge`,
    `Air conditioning`,
    `Baby seat`,
    `Breakfast`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16
  },
  id: 1,
  isPremium: false,
  isFavorite: false,
  maxAdults: 4,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`
  ],
  title: `Amazing and Extremely Central Flat`,
  rating: 4.3,
  type: `APARTMENT`,
  bedrooms: 3,
  price: 323,
  goods: [
    `Towels`,
    `Baby seat`,
    `Laptop friendly workspace`,
    `Dishwasher`,
    `Washing machine`,
    `Air conditioning`,
    `Washer`,
    `Coffee machine`,
    `Breakfast`,
    `Fridge`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `This is a place for dreamers to reset, reflect, and create. Designed with a \`slow\` pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.`,
  location: {
    latitude: 50.839557,
    longitude: 4.346697,
    zoom: 16
  },
  id: 2,
  isPremium: false,
  isFavorite: false,
  maxAdults: 8,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`
},
{
  city: {
    name: `Brussels`,
    location: {
      latitude: 50.846557,
      longitude: 4.351697,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`
  ],
  title: `The house among olive `,
  rating: 4.6,
  type: `APARTMENT`,
  bedrooms: 4,
  price: 218,
  goods: [
    `Washer`,
    `Laptop friendly workspace`,
    `Breakfast`,
    `Air conditioning`,
    `Baby seat`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
  location: {
    latitude: 50.865556999999995,
    longitude: 4.371696999999999,
    zoom: 16
  },
  id: 3,
  isPremium: false,
  isFavorite: false,
  maxAdults: 7,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
}
];

export const offer = {
  city: {
    name: `Cologne`,
    location: {
      latitude: 50.938361,
      longitude: 6.959974,
      zoom: 13
    }
  },
  images: [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`
  ],
  title: `The house among olive `,
  rating: 3.8,
  type: `HOUSE`,
  bedrooms: 3,
  price: 493,
  goods: [
    `Laptop friendly workspace`,
    `Washer`,
    `Towels`,
    `Fridge`,
    `Air conditioning`,
    `Baby seat`,
    `Breakfast`
  ],
  host: {
    avatarUrl: `img/avatar-angelina.jpg`,
    isPro: true
  },
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  location: {
    latitude: 50.957361,
    longitude: 6.9509739999999995,
    zoom: 16
  },
  id: 1,
  isPremium: false,
  isFavorite: false,
  maxAdults: 4,
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`
};

export const favoriteIds = [13, 15, 24, 26];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const activeCity = `Paris`;

export const activeSortType = `Popular`;

export const review = {
  id: 1,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    isPro: false
  },
  rating: 2,
  comment: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
  date: `2020-10-28T13:38:44.753Z`
};

export const reviews = [{
  id: 1,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    isPro: false
  },
  rating: 2,
  comment: `The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.`,
  date: `2020-10-28T13:38:44.753Z`
},
{
  id: 2,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
    isPro: false
  },
  rating: 3,
  comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
  date: `2020-10-28T13:38:44.753Z`
},
{
  id: 3,
  user: {
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/2.jpg`,
    isPro: false
  },
  rating: 4,
  comment: `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
  date: `2020-10-28T13:38:44.753Z`
}
];

export const currentClasses = {
  listClass: `near-places__list`,
  cardClass: `near-places__card`,
  imgClass: `near-places__image-wrapper`
};

export const match = {
  path: `/offer/:id`,
  url: `/offer/11`,
  isExact: true,
  params: {
    id: `11`
  }
};

export const mapStyle = {
  display: `flex`,
  height: `100%`,
  width: 1144,
  margin: `0 auto`
};

export const offersId = [1, 2, 3];
