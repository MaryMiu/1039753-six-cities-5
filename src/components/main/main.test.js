import React from "react";
import renderer from "react-test-renderer";
import {Main} from "./main";
import {AuthorizationStatus, activeCity, offers, activeSortType} from "../../mock/for-test";

const noop = () => {};

jest.mock(`../header/header`, () => `Header`);
jest.mock(`../places-list/places-list`, () => `PlacesList`);
jest.mock(`../sortlist/sortlist`, () => `Sortlist`);
jest.mock(`../map/map`, () => `Map`);
jest.mock(`../menu/menu`, () => `Menu`);
jest.mock(`../main-empty/main-empty`, () => `MainEmpty`);

describe(`Main render correctly`, () => {

  it(`Should Main render correctly with offers`, () => {
    const tree = renderer
    .create(<Main
      authorizationStatus={AuthorizationStatus.AUTH}
      onSubmit={noop}
      activeCity={activeCity}
      offers={offers}
      activeSortType={activeSortType}
      fetchOfferListAction={noop}
    />)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Main render correctly without offers`, () => {
    const tree = renderer
    .create(<Main
      authorizationStatus={AuthorizationStatus.AUTH}
      onSubmit={noop}
      activeCity={activeCity}
      offers={[]}
      activeSortType={activeSortType}
      fetchOfferListAction={noop}
    />)
    .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
