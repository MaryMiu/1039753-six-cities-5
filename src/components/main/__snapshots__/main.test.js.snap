// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main render correctly Should Main render correctly with offers 1`] = `
<div
  className="page page--gray page--main "
>
  <Header />
  <main
    className="page__main page__main--index "
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <Menu />
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            3
             places to stay in 
            Paris
          </b>
          <Sortlist />
          <PlacesList
            currentClasses={
              Object {
                "cardClass": "cities__place-card",
                "imgClass": "cities__image-wrapper",
                "listClass": "cities__places-list tabs__content",
              }
            }
            offers={
              Array [
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 50.938361,
                      "longitude": 6.959974,
                      "zoom": 13,
                    },
                    "name": "Cologne",
                  },
                  "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                  "goods": Array [
                    "Laptop friendly workspace",
                    "Washer",
                    "Towels",
                    "Fridge",
                    "Air conditioning",
                    "Baby seat",
                    "Breakfast",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "isPro": true,
                  },
                  "id": 1,
                  "images": Array [
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 50.957361,
                    "longitude": 6.9509739999999995,
                    "zoom": 16,
                  },
                  "maxAdults": 4,
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                  "price": 493,
                  "rating": 3.8,
                  "title": "The house among olive ",
                  "type": "HOUSE",
                },
                Object {
                  "bedrooms": 3,
                  "city": Object {
                    "location": Object {
                      "latitude": 50.846557,
                      "longitude": 4.351697,
                      "zoom": 13,
                    },
                    "name": "Brussels",
                  },
                  "description": "This is a place for dreamers to reset, reflect, and create. Designed with a \`slow\` pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                  "goods": Array [
                    "Towels",
                    "Baby seat",
                    "Laptop friendly workspace",
                    "Dishwasher",
                    "Washing machine",
                    "Air conditioning",
                    "Washer",
                    "Coffee machine",
                    "Breakfast",
                    "Fridge",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "isPro": true,
                  },
                  "id": 2,
                  "images": Array [
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 50.839557,
                    "longitude": 4.346697,
                    "zoom": 16,
                  },
                  "maxAdults": 8,
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                  "price": 323,
                  "rating": 4.3,
                  "title": "Amazing and Extremely Central Flat",
                  "type": "APARTMENT",
                },
                Object {
                  "bedrooms": 4,
                  "city": Object {
                    "location": Object {
                      "latitude": 50.846557,
                      "longitude": 4.351697,
                      "zoom": 13,
                    },
                    "name": "Brussels",
                  },
                  "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                  "goods": Array [
                    "Washer",
                    "Laptop friendly workspace",
                    "Breakfast",
                    "Air conditioning",
                    "Baby seat",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "isPro": true,
                  },
                  "id": 3,
                  "images": Array [
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                    "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": false,
                  "location": Object {
                    "latitude": 50.865556999999995,
                    "longitude": 4.371696999999999,
                    "zoom": 16,
                  },
                  "maxAdults": 7,
                  "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                  "price": 218,
                  "rating": 4.6,
                  "title": "The house among olive ",
                  "type": "APARTMENT",
                },
              ]
            }
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <Map
              mapStyle={
                Object {
                  "display": "flex",
                  "height": "100%",
                  "width": "100%",
                }
              }
              offers={
                Array [
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 50.938361,
                        "longitude": 6.959974,
                        "zoom": 13,
                      },
                      "name": "Cologne",
                    },
                    "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                    "goods": Array [
                      "Laptop friendly workspace",
                      "Washer",
                      "Towels",
                      "Fridge",
                      "Air conditioning",
                      "Baby seat",
                      "Breakfast",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "isPro": true,
                    },
                    "id": 1,
                    "images": Array [
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 50.957361,
                      "longitude": 6.9509739999999995,
                      "zoom": 16,
                    },
                    "maxAdults": 4,
                    "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                    "price": 493,
                    "rating": 3.8,
                    "title": "The house among olive ",
                    "type": "HOUSE",
                  },
                  Object {
                    "bedrooms": 3,
                    "city": Object {
                      "location": Object {
                        "latitude": 50.846557,
                        "longitude": 4.351697,
                        "zoom": 13,
                      },
                      "name": "Brussels",
                    },
                    "description": "This is a place for dreamers to reset, reflect, and create. Designed with a \`slow\` pace in mind, our hope is that you enjoy every part of your stay; from making local coffee by drip in the morning, choosing the perfect record to put on as the sun sets.",
                    "goods": Array [
                      "Towels",
                      "Baby seat",
                      "Laptop friendly workspace",
                      "Dishwasher",
                      "Washing machine",
                      "Air conditioning",
                      "Washer",
                      "Coffee machine",
                      "Breakfast",
                      "Fridge",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "isPro": true,
                    },
                    "id": 2,
                    "images": Array [
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/9.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 50.839557,
                      "longitude": 4.346697,
                      "zoom": 16,
                    },
                    "maxAdults": 8,
                    "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                    "price": 323,
                    "rating": 4.3,
                    "title": "Amazing and Extremely Central Flat",
                    "type": "APARTMENT",
                  },
                  Object {
                    "bedrooms": 4,
                    "city": Object {
                      "location": Object {
                        "latitude": 50.846557,
                        "longitude": 4.351697,
                        "zoom": 13,
                      },
                      "name": "Brussels",
                    },
                    "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                    "goods": Array [
                      "Washer",
                      "Laptop friendly workspace",
                      "Breakfast",
                      "Air conditioning",
                      "Baby seat",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "isPro": true,
                    },
                    "id": 3,
                    "images": Array [
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/2.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/1.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/12.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/17.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/5.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/11.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/4.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/10.jpg",
                      "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": false,
                    "location": Object {
                      "latitude": 50.865556999999995,
                      "longitude": 4.371696999999999,
                      "zoom": 16,
                    },
                    "maxAdults": 7,
                    "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
                    "price": 218,
                    "rating": 4.6,
                    "title": "The house among olive ",
                    "type": "APARTMENT",
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;

exports[`Main render correctly Should Main render correctly without offers 1`] = `
<div
  className="page page--gray page--main page__main--index-empty"
>
  <Header />
  <main
    className="page__main page__main--index cities__places-container--empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <Menu />
    </div>
    <div
      className="cities"
    >
      <MainEmpty />
    </div>
  </main>
</div>
`;
